### DISCUSSION TITLE
Searches only primary monitor?

### DISCUSSION THREAD

**Discussion by Pervect66** [Spawnova/ShinsImageScanClass#10] [Q&A]
I used the example to find an image, but it only seems to detect it if it's on the primary monitor (I use four):

#include ShinsImageScanClass.ahk
SetBatchLines -1
ListLines Off
Process, Priority, , A

scan := New ShinsImageScanClass()

Loop {
if (scan.Image("alarm.png")) {
SoundPlay, alarm.mp3 , Wait
}

Is there a way to have it search all monitors?

**Comment by Spawnova** [comment]
Hi there, I've just updated the class to support multi monitor setup, it just enables it to work for now, I will try and add performance improvements later when using it this way

**Reply by Pervect66** [reply]
Excellent, that works! Does have a huge CPU penalty though..... More than AHK's ImageSearch. That has a 5 or 6% CPU util searching 4 monitors constantly. This more like 15%?

**Reply by Spawnova** [reply]
Does the default ahk image search support the entire virtual screen? Assuming you are using 4 1080p monitors you are searching a 7680x4320 screen for the image, even more if your monitor layout does not line up into a square or rectangle.
I've updated the class once more this time improving how the buffer updates on desktop regions, which should performance gain when using any region functions, which you should do if possible.

**Comment by Pervect66** [comment]
I used to have 2x2 1080p, and one to the right of those.... five total: I made that work :-)
Now only  2x2 1080p.

```
Loop {
CoordMode, Screen
SysGet, VirtualWidth, 78
SysGet, VirtualHeight, 79
ImageSearch,,, 0, 0, %VirtualWidth%, %VirtualHeight%, alarm.png
if (ErrorLevel = 0) 
SoundPlay, alarm.mp3 , Wait
}
```

I'll test this version and let you know! Thanks a lot for a great library!
